sdk/search/Azure.Search.Documents/tests/Batching/BatchingTests.cs

```diff
@@ -389,10 +393,20 @@ public async Task Champion_BasicCheckpointing()
             await indexer.UploadDocumentsAsync(data.Skip(500));
 
             await DelayAsync(TimeSpan.FromSeconds(5), TimeSpan.FromMilliseconds(250));
-            Assert.AreEqual(1001 - BatchSize, pending.Count);
+
+            Assert.AreEqual(0, removeFailedCount);
+            if ((pending.Count == 1001) && (Mode != RecordedTestMode.Playback))
+            {
+                Assert.Inconclusive("Publishing has not started.");
```

Comment:
What does "publishing has not started" mean?

Reply:
That a batch with these documents has not yet been submitted for processing.

Reply:
Can/should we force that by flushing?

Reply:
That would change the intent of the test IMO.

We are testing that with `AutoFlush` enabled and with 0 delay (`AutoFlushInterval` == null) -
1. Submitting more than (default) `BatchSize` number of actions triggers a submission
2. "Total count - `BatchSize`" number of actions stay unsubmitted in the pending queue afterwards
3. Explicit flushing takes care of remaining items

If we add a flush after [1] and before verifying [2], we are no longer testing the auto-submission behavior.

Reply:
Then why does publishing not having started mean the test is inconclusive rather than failed?  We just don't want to add a longer wait?

Reply:
Ok, I'll do that.